components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java

```diff
@@ -1309,6 +1330,223 @@ public void addApplicationWithTemplateIdAndTemplateVersionData(String templateId
                 REGISTRY_SYSTEM_USERNAME);
     }
 
+    @DataProvider(name = "applicationCertificateDataProvider")
+    public Object[][] applicationCertificateDataProvider() {
+
+        ServiceProvider inputSP = new ServiceProvider();
+        inputSP.setApplicationName(APPLICATION_NAME_1);
+        inputSP.setCertificateContent(CERTIFICATE);
+
+        Certificate certificate = new Certificate.Builder()
+                .id(String.valueOf(CERTIFICATE_ID))
+                .name(APPLICATION_NAME_1)
+                .certificateContent(CERTIFICATE)
+                .build();
+
+        return new Object[][] {
+                {inputSP, certificate}
+        };
+    }
+
+    @Test(dataProvider = "applicationCertificateDataProvider")
+    public void testApplicationCertificateSuccess(ServiceProvider inputSP, Certificate certificate)
```

Comment:
please break to different methods. Is there any specific reason to combine these multiple tests ?

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6087/commits/9a2dc889d25ec1c2d6d950643a6310f1717fd44a